server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -906,11 +907,11 @@ public Query createKnnQuery(
                     }
                     yield parentFilter != null
                         ? new DiversifyingChildrenByteKnnVectorQuery(name(), bytes, filter, numCands, parentFilter)
-                        : new KnnByteVectorQuery(name(), bytes, numCands, filter);
+                        : new ProfilingKnnByteVectorQuery(name(), bytes, numCands, filter);
                 }
                 case FLOAT -> parentFilter != null
                     ? new DiversifyingChildrenFloatKnnVectorQuery(name(), queryVector, filter, numCands, parentFilter)
```

Comment:
We should profile these diversifying as well.

Reply:
Updated